Section: Overview of game theoretic model

MichalisPanayides opened the pull request.

MichalisPanayides committed:
Write initial draft of game overview section

MichalisPanayides committed:
Add texlive-latex-extra to ci.yml

Update the CI so that it installs the texlive-latex-extra package that
includes the pgf package that tikz requires in order to work

MichalisPanayides committed:
Apply minor changes and fix typos

MichalisPanayides committed:
Finish writing of section and add referecne

MichalisPanayides committed:
Update known.py

MichalisPanayides committed:
Minor change

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
\ No newline at end of file
+\section{Overview of game theoretic model}
+
+The problem studied is a 3-player game theoretic scenario between two queueing 
+systems and a service that distributes individuals to them. 
```
```suggestion
The problem studied is a 3-player normal form game.

 The players are:
  
 \begin{itemize}
 
      \item The decision makers of two queueing 
systems;
      \item a service that distributes individuals to these two queueing systems.
 \end{itemize}
 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+
+The problem studied is a 3-player game theoretic scenario between two queueing 
+systems and a service that distributes individuals to them. 
+Each player in this game has its own objectives and cares about meeting its own 
+targets.
```
```suggestion
This is a standard Normal form game~\cite{},  % TODO Cite a Game Theory text book (Maschler et al. would do)
in that each player in this game has their own objectives which they aim to optimise.
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+Each player in this game has its own objectives and cares about meeting its own 
+targets.
+More specifically, the queueing systems' objective is captured by an upper bound
+of the time that a fixed proportion of individuals need to spend in the system, 
```
```suggestion
of the time that a fixed proportion of individuals  spend in the system, 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+More specifically, the queueing systems' objective is captured by an upper bound
+of the time that a fixed proportion of individuals need to spend in the system, 
+while the distributor aims to minimise the time that its individuals 
+are blocked.  
```
```suggestion
are blocked.   % TODO Depending on how this is described earlier, we might need a little more here (as to what we mean by "blocked")
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+other individuals from other sources. 
+Each queueing system may then choose to block the individuals that arrive from 
+the distributor when the system reaches a certain capacity. 
+Thus, the set of strategies that they may choose to play in the game are defined 
+as the set of possible choices that the two queueing systems have. 
+In other words, each queueing system is given \( N_A \) and \( N_B \) number of 
+strategies where \( N_A \) and \( N_B \) are the total capacities of the two 
+queueing systems.
+The \textit{threshold} parameter is thus used to capture the chosen 
+strategy for each hospital where \( T_A \in \left[1, N_A \right] \) and 
+\( T_B \in \left[1, N_B \right] \)
```
```suggestion
The strategy sets for each queueing system is the set \(\{T \in \mathbb{N} \;|\; 1 \leq T \leq N \) and \( N_B \) 
where \(N\in\{N_A, N_B\}\) are the total capacities of the two queueing systems. We denote the chosen actions from the strategy set as \(T_A, T_B\) and call these \textit{threshold}s.
```

MichalisPanayides replied:
That is looks definitely better.

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+\( T_B \in \left[1, N_B \right] \)
+Their choice of strategy will then rely solely on satisfying their own 
+objective, which is to make sure that the waiting time of a proportion of 
+individuals will be below the predefined target time.
+
```
```suggestion
individuals will be below the predefined target time.

This is shown diagrammatically in Figure~ref{fig:diagram_of_queueing_system}.

% TODO Draw diagram of queueing system
% This should include the parameters
% and clearly show the decision variable.
% Note: only do this for a single queueing system.

```

MichalisPanayides replied:
By diagram do you mean something like the general queuing diagram shown below or the Markov chain one?

![image](https://user-images.githubusercontent.com/20097997/114890543-058ab900-9e03-11eb-8ba8-4be77e394505.png)

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+    P(W < \textit{ Target}) \geq \hat{P}
+\end{equation}
+
+where \(W\) is the waiting time of all individuals, \textit{Target} is the time 
```
```suggestion
where \(W\) is the mean waiting time of all individuals, \textit{Target} is the time 
```

I'd not use `Target` in text at all (if you do it should not be in italics), just find some letter you haven't used yet and define that as the target.

MichalisPanayides replied:
Hmmm🤔. Yeah, that makes sense. Will use `R` instead then 👍

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+individuals will be below the predefined target time.
+
+\begin{equation}
+    P(W < \textit{ Target}) \geq \hat{P}
```
```suggestion
    P(W < \text{ Target}) \geq \hat{P}
```

Italics within a math environment look like you're multiplying T by a by r by g by e by t.

MichalisPanayides replied:
Will keep in mind 👍

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+target. 
+There are numerous objective functions that can be used to capture this 
+behaviour. 
+The first one would be to use the threshold that maximises the probability that 
```
```suggestion
For example one approach is to use the threshold that maximises the probability that 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+    \arg \max_{T_i} \quad P(W_i < \textit{Target})
+\end{equation}
+
+Perhaps a more sophisticated objective function would be to get the proportion 
+of individuals as close to the percentage aim. 
+In other words, to find the threshold that minimises the difference between the 
+probability and the percentage goal (or maximise its negation).
```
```suggestion
A more sophisticated objective function would be to get the proportion 
of individuals as close to the percentage aim. 
In other words, to find the threshold that minimises the difference between the 
probability and the percentage goal (or maximise its negation).
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+\end{equation}
+
+
+On the other side of the spectrum, the distributor has its own choices 
+to make and its own goals to satisfy.
```
```suggestion
The third player, the distributor has their own choices 
to make and theirs own goals to satisfy.
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+
+On the other side of the spectrum, the distributor has its own choices 
+to make and its own goals to satisfy.
+The set of strategies it has can be defined as the different ways that it can 
+distribute individuals to the two queueing system.
+Therefore, unlike the queueing systems, the distributor has a 
+continuous set of strategies that goes from \( 0\% - 100\% \). 
+For example, it may choose to send \(35\%\) of its individuals to the first 
+queueing system and \(65\%\) to the second one.
```
```suggestion
The strategy set of the third player is the proportion \(0 \leq p \leq 1\) of individuals it sends to the first queueing system (the proportion \(1 - q\) is sent to the second queueing system).
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+Thus, its objective is to minimise the blocked time of the individuals 
+that they send to the two queueing systems.
+Apart from the time being blocked, an additional aspect that may affect the 
+decision of the distribution service is the proportion of lost individuals.
```
We need to clarify above (somewhere near the diagram of the queueing system that individuals can be lost at all).

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+    \alpha P(L_B) + (1 - \alpha) B_B
+\end{equation}
+
+Here, the purpose of \(\alpha\) is to tune the ``importance'' of each objective,
+where high \(\alpha\) indicates a higher weight on the proportion of lost 
+individuals and smaller \(\alpha\) a higher weight on the time blocked. 
```
```suggestion
Here, \(\alpha\) represents the ``importance'' of each objective,
where high \(\alpha\) indicates a higher weight on the proportion of lost 
individuals and smaller \(\alpha\) a higher weight on the time blocked. 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+individuals and smaller \(\alpha\) a higher weight on the time blocked. 
+
+
+Using equations \ref{eq:obj-queueing-systems} and \ref{eq:obj-distributor}, an
+imperfect information extensive form game may arise. 
```
```suggestion
Using equations \ref{eq:obj-queueing-systems} and \ref{eq:obj-distributor gives an
imperfect information extensive form game.
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+An imperfect information game is defined as an extensive form game where some 
+of the information about the game state is hidden for at least one of the 
+players \cite{Berwanger2008}. In this study the state of the problem that is
+hidden is the threshold that each player chooses to play.
```
```suggestion
hidden is the threshold that each of the first two players chooses to play.
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+hidden is the threshold that each player chooses to play.
+In other words, each queueing system chooses to play a strategy without the 
+knowing the other system's strategy.
+The distributor then, fully aware of the chosen strategies are, distributes 
```
```suggestion
The distributor then, fully aware of the chosen threshold strategies, distributes 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+    \label{fig:imperfect-info-game}
+\end{figure}
+
+Figure \ref{fig:imperfect-info-game} illustrates such a game works. The first 
```
```suggestion
The first 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+knowing the other system's strategy.
+The distributor then, fully aware of the chosen strategies are, distributes 
+individuals among the two systems in order to minimise the time that its 
+individuals will be blocked. 
```
```suggestion
individuals will be blocked. Figure \ref{fig:imperfect-info-game} illustrates this. 
```

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,168 @@
+\end{equation}
+
+Based on the choice of strategy of these two players the distributor will then 
+make its own choice of the proportion of individuals to send to each system.
```
```suggestion
make their own choice of the proportion of individuals to send to each system.
```

MichalisPanayides committed:
Apply suggestions from code review

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

MichalisPanayides committed:
Apply suggestions from review

Rewording some sentences, restructure some paragraphs and add several
TODOs.

MichalisPanayides committed:
Add reference for normal form games

MichalisPanayides committed:
Update known.py file

MichalisPanayides committed:
Add diagram of the queueing model

Added the diagram for the queueing model and some text to
accompany it. There were also minor changes to some of the
prewritten text.

MichalisPanayides committed:
Minor changes in queueing diagram

Add a gap between the lines of the second queue and add T in that gab

drvinceknight commented on sections/model_overview/main.tex:
```diff
@@ -1 +1,249 @@
+        % Arrow line to Queue 2
+        \draw[->] (4.2, 0.525) -- (4.8, 0.525);  
+    \end{tikzpicture}
+    \caption{Queueing Model}
```
```suggestion
    \caption{A diagrammatic representation of the queueing model. 
    The threshold \(T\) only applies to arrivals from the first buffer. 
    If the second buffer is at that threshold only individuals of the 
    first type are accepted (at a rate \(\lambda_1\)) and individuals 
    of the second type (arriving at a rate \(\lambda_2\)) are held 
    in the blocked in the first buffer.}
```

drvinceknight replied:
Once that's in, let's merge this PR (and rebase #5)

MichalisPanayides committed:
Change caption of figure
